File: cha-parser/tests/fixtures/simple.ts
import { readFile } from "fs";
import path from "path";

function greet(name: string): string {
    return `Hello, ${name}`;
}

class Greeter {
    name: string;

    constructor(name: string) {
        this.name = name;
    }

    greet(): string {
        return `Hello, ${this.name}`;
    }
}

File: cha-parser/tests/fixtures/simple.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int x;
    int y;
} Point;

struct Color {
    int r;
    int g;
    int b;
};

int add(int a, int b) {
    return a + b;
}

void greet(const char* name) {
    printf("Hello, %s!\n", name);
}
